io/printer.go

```diff
@@ -6,10 +6,15 @@ type Printer interface {
 	Println(string, ...color.Attribute)
 }
 
+var printerInstance Printer
+
 type printer struct{}
 
-func NewPrinter() Printer {
-	return &printer{}
+func GetPrinter() Printer {
```

Comment:
Can we get rid of this singleton since in the CLI it won't have any performance effects? If need be, we can inject printer at the top level like injecting config.